### ISSUE TITLE
Ubuntu 24.04 is now available

### ISSUE THREAD

**Issue by mikhailkoliada** [actions/runner-images#9848]
### Breaking changes

Ubuntu 24.04 is now available for all GitHub Actions and Azure DevOps users.
You can use the ubuntu 24.04 image label in your YAML to select this image.

**GitHub Actions**
```
jobs:
  jobName:
    runs-on: ubuntu-24.04
```

**Azure DevOps**
```
pool:
   name: 'Azure Pipelines'
   vmImage: 'ubuntu-24.04'
```

The image is marked as "beta" for now. It means some software can be unstable on the new platform. Also, there could be queueing issues as the capacity will be balanced only throughout the next weeks.
**Please report any problems with the new image to this repository.**

### Platforms affected

- [X] Azure DevOps
- [X] GitHub Actions

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [X] Ubuntu 24.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Software difference between Ubuntu 22 and Ubuntu 24

| Tool name | Ubuntu 22.04 | Ubuntu 24.04 | Notes |
|-----------|--------------|--------------|-------|
| Clang | <ul><li>13.* </li><li>14.* (default)</li><li>15.* </li></ul> | <ul><li>16.* </li><li>17.* </li><li>18.* (default)</li></ul> | The most recent versions are installed |
| GCC / GNU C++ / GNU Fortran | <ul><li>9.* </li><li>10.* </li><li>11.* </li><li>12.* </li><li>13.* </li></ul> | <ul><li>12.* </li><li>13.* </li><li>14.* </li></ul> | The most recent versions are installed |
| Kotlin | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| PHP | 8.1.* | 8.3.* | The most recent version are installed |
| Swift | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Java | <ul><li>8.* </li><li>11.* (default)</li><li>17.* </li><li>21.* </li></ul> | <ul><li>8.* </li><li>11.* </li><li>17.* (default)</li><li>21.* </li></ul> | Default Java switched to 17.* for Ubuntu 24.04 image. |
| Python | <ul><li>3.7.* (cached)</li><li>3.8.* (cached)</li><li>3.9.* (cached)</li><li>3.10.* (default)</li><li>3.11.* (cached)</li><li>3.12.* (cached)</li></ul> | <ul><li>3.9.* (cached)</li><li>3.10.* (default)</li><li>3.11.* (cached)</li><li>3.12.* (default)</li></ul> | Pre-cached versions currently unavailable. Default version switched to the latest one. On GitHub Actions, [actions/setup-python](https://github.com/actions/setup-python) can install any version on-flight so this change doesn't impact users |
| Go | <ul><li>1.20.* (cached)</li><li>1.21.* (default)</li><li>1.22.* (cached)</li></ul> | <ul><li>1.20.* (cached)</li><li>1.21.* (cached)</li><li>1.22.* (default)</li></ul> | If your use-case requires using any of these versions, consider using tasks to install Go on-flight: <ul><li>[actions/setup-go](https://github.com/actions/setup-go) (GitHub Actions) </li><li>[Go Tool Installer](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/go-tool?view=azure-devops) (Azure DevOps)</li></ul> |
| PyPy | <ul><li>3.7.* (cached)</li><li>3.8.* (cached)</li><li>3.9.* (cached)</li><li>3.10.* (cached)</li></ul> | <ul><li>3.9.* (cached)</li><li>3.10.* (cached)</li></ul> | Deprecated all the versions besides the most recent ones |
| Ruby | <ul><li>3.0.* (default)</li><li>3.1.* (cached)</li></ul> | <ul><li>3.2.* (default)</li></ul> | Pre-cached versions currently unavailable. Default version switched to the latest one. |
| Node.js | <ul><li>16.* (cached)</li><li>18.* (default)</li><li>20.* (cached)</li></ul> | <ul><li>16.* (cached)</li><li>18.* (cached)</li><li>20.* (default)</li></ul> | If your use-case requires using any of these versions, consider using tasks to install Node.js on-flight: <ul><li>[actions/setup-node](https://github.com/actions/setup-node) (GitHub Actions) </li><li>[Node.js Tool Installer](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/node-js?view=azure-devops) (Azure DevOps)</li></ul> |
| Heroku | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Leiningen | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Mono / MSBuild / NuGet | latest available | - | Software is not available for Ubuntu 24 at the moment. |
| Packer / Terraform | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| R | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| SVN | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Alibaba Cloud CLI | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Google Cloud CLI | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Netlify CLI | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| OpenShift CLI | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| ORAS CLI | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Vercel CLI | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Bindgen / Cbindgen | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Cargo audit/clippy/outdated | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Microsoft Edge | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Microsoft Edge WebDriver | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Mozilla Firefox | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Mozilla Geckodriver | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| .NET Core SDK | <ul><li>6.* </li><li>7.* </li><li>8.* </li></ul> | <ul><li>8.* </li></ul> | Please consider using tasks to install any version on-flight: <ul><li>[actions/setup-dotnet](https://github.com/actions/setup-dotnet) (GitHub Actions) </li><li>[Use .NET Core](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops) (Azure DevOps)</li></ul> |
| PostgreSQL | 14.* | 16.* | More recent version are installed |
| MS SQL Server Client Tools | sqlcmd / SqlPackage | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| MarkdownPS Module | latest available | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |
| Android Command Line Tools | 9.0 | 12.0 | The most recent version are installed |
| Android SDK Build-tools | <ul><li>34.0.0 </li><li>33.0.0 </li><li>33.0.1 </li><li>33.0.2 </li><li>33.0.3 </li><li>32.0.0 </li><li>31.0.0</li></ul> | <ul><li>34.0.0</li></ul> | The most recent version are installed |
| Android NDK | <ul><li>25.* (default)</li><li>26.* </li></ul> | <ul><li>26.* (default)</li></ul> | The most recent version are installed |
| Cached Docker images | <ul><li>alpine:3.16</li><li>alpine:3.17</li><li>alpine:3.18</li><li>debian:10</li><li>debian:11</li><li>moby/buildkit:latest</li><li>node:16</li><li>node:16-alpine</li><li>node:18</li><li>node:18-alpine</li><li>node:20</li><li>node:20-alpine</li><li>ubuntu:20.04</li><li>ubuntu:22.04</li></ul> | - | Removed from the Ubuntu 24.04 image due to maintenance reasons. |

**Comment by MikeMcC399** [comment]
@mikhailkoliada 

Thanks very much for the announcement here!

Will you mark it as beta in the [README](https://github.com/actions/runner-images/blob/main/README.md)?

![image](https://github.com/actions/runner-images/assets/66998419/3d787ee5-d0ef-48a1-b07d-6db4a1ad3574)

**Comment by mikhailkoliada** [comment]
@MikeMcC399 yep, I forgot about it, now done :)

**Comment by esaulpaugh** [comment]
Run java -version
  java -version
  chmod +x ./gradlew
  ./gradlew build fatJar --no-daemon
  java -jar build/libs/headlong-cli-1.1-SNAPSHOT.jar -version
  shell: /usr/bin/bash -e {0}
  env:
    GRADLE_ACTION_ID: gradle/actions/wrapper-validation
    JAVA_HOME: /opt/hostedtoolcache/Java_Zulu_jdk/8.0.41[2](https://github.com/esaulpaugh/headlong-cli/actions/runs/9086407707/job/24972106707#step:5:2)-8/x86
    JAVA_HOME_8_X86: /opt/hostedtoolcache/Java_Zulu_jdk/8.0.412-8/x86
The futex facility returned an unexpected error code.
/home/runner/work/_temp/f9c9c8[3](https://github.com/esaulpaugh/headlong-cli/actions/runs/9086407707/job/24972106707#step:5:3)d-d717-4c5b-9083-6922b5f9175c.sh: line 1:  2720 Aborted                 (core dumped) java -version
Error: Process completed with exit code 13[4](https://github.com/esaulpaugh/headlong-cli/actions/runs/9086407707/job/24972106707#step:5:4).

**Comment by andreasabel** [comment]
`convert` (imagemagick) is no longer pre-installed.
- https://github.com/agda/agda/actions/runs/9086817771/job/24973341288?pr=7261#step:7:105

**Comment by edmorley** [comment]
@mikhailkoliada Thank you for the release! What is the timeline for updating the `ubuntu-latest` alias to default to Ubuntu 24.04 instead? (Context: We would like to upgrade to the latest Ubuntu version, however, it will save a lot of churn of replacing `ubuntu-latest` with `ubuntu-24.04` if we could instead wait for the default to change.)

**Comment by MikeMcC399** [comment]
@edmorley 

In https://github.com/actions/runner-images/issues/9691#issuecomment-2063926929 there was mention of a date in August 2024 for GA. That might coincide with the change you are asking about.

**Comment by kojoru** [comment]
Are the changes marked "Removed from the Ubuntu 24.04 image due to maintenance reasons." final? Is there a particular common reason? Can we somehow vote/plead to revert some/any of those choices?

**Comment by mikhailkoliada** [comment]
@kojoru there will definitely be changes until GA including the software list installation (for instance we are to add firefox & edge, but a bit later), but we do not guarantee we are to address all the requests, everybody is welcome to file new tools addition requests and we'll review them.

**Comment by mikhailkoliada** [comment]
@edmorley migrating to lates is a different process that usually takes time, 24.04 will eventually become latest after its GA

**Comment by adelton** [comment]
@mikhailkoliada Would you like to amend the software difference table in your first post with the information that Podman gets updated from Podman 3.4.4 in the Ubuntu 22.04 image to Podman 4.9.3 in 24.04?

**Comment by sanderdelfos** [comment]
@mikhailkoliada Cannot seem to use the 24.04 image:
![image](https://github.com/actions/runner-images/assets/134610142/8869b3a0-7647-481a-a467-5ab2ba83f5ee)

**Comment by MikeMcC399** [comment]
@sanderdelfos 

I noticed also that `ubuntu-24.04` does not yet appear on the [Azure Pipelines > Microsoft-hosted agents](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted) list.

**Comment by sanderdelfos** [comment]
@MikeMcC399 True, but it is listed in this issue as ready for use and also listed here:
https://github.com/actions/runner-images

**Comment by dandrestor** [comment]
I noticed `Ubuntu2404` was not added as an option to `helpers/GenerateResourcesAndImage.ps1`. Are you holding off until GA or are there any plans to do that earlier?

**Comment by simonmichael** [comment]
Haskell tools ([2204](https://github.com/actions/runner-images/blob/ubuntu22/20240514.2/images/ubuntu/Ubuntu2204-Readme.md#haskell-tools), [2404](https://github.com/actions/runner-images/blob/ubuntu22/20240514.2/images/ubuntu/Ubuntu2404-Readme.md#haskell-tools)) are not mentioned in the "Software differences" list above. (Thanks for these, and for the new image!)

**Comment by f-lawe** [comment]
@pmandadkes1207 It seems that Ubuntu 24.04 is not fully deployed on Azure DevOps, @sanderdelfos and I can't even use it on a YAML pipeline (but you can). Let's give it a bit more time :)

**Comment by erik-bershel** [comment]
### Heads up! 📯 📯 📯 
#### Couple updates regarding Ubuntu-24.04 status and updates.

- The release in ADO, to our great regret, is indeed delayed, but this should not drag on for long. Please check the page describing available [Microsoft Hosted agents](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software) periodically - changes will be reflected there.
- The diff table includes either those packages and programs which versions we control manually, rather than providing the "latest" available version every time, or packages that are the most critical (for example, the default versions for `Ruby` and `Python`). The most complete state of the image can be found [here](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md). And please keep in mind that the situation will change quite quickly, since the availability of packages at the start of a new image changes most quickly.
- We will be happy to investigate every comment and suggestion regarding the content of the new image, although we understand that we will not be able to satisfy every such request.

**Comment by iXyles** [comment]
Over the day I started to run into an issue in azure devops regarding "azure cli" after this ubuntu latest build, with the task `AzureResourceManagerTemplateDeployment@3`

```
Starting: AzureResourceManagerTemplateDeployment
==============================================================================
Task         : ARM template deployment
Description  : Deploy an Azure Resource Manager (ARM) template to all the deployment scopes
Version      : 3.238.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/azure-resource-group-deployment
==============================================================================
ARM Service Connection deployment scope - Subscription
Checking if the following resource group exists: **REDACTED**.
Resource group exists: true.
Creating deployment parameters.
##[error]Check out the troubleshooting guide to see if your issue is addressed: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-resource-group-deployment?view=azure-devops#troubleshooting
##[error]Error: Command failed: az version
/bin/sh: 1: az: not found

Finishing: AzureResourceManagerTemplateDeployment
```

Is this an expected thing to be after the release? Or an unexpected breaking change?

**Comment by erik-bershel** [comment]
Hey @iXyles!

`ubuntu-24.04` <> `ubuntu-latest`
Your issue is related to `ubuntu22.04`.

**Comment by erik-bershel** [comment]
@iXyles you may file a compatibility issue here: https://github.com/microsoft/azure-pipelines-tasks - it's a task owners repository. We won't accept this bug since base functionality of `azure cli` available. See there:
<img width="1065" alt="Screenshot 2024-05-17 at 16 45 09" src="https://github.com/actions/runner-images/assets/110455084/cd48ba96-8929-43e0-968b-817b7acc2fc1">

**Comment by iXyles** [comment]
Massive thanks @sanderdelfos for taking the time to verify this, will look into doing that then!

**Comment by mkniewallner** [comment]
Would it be possible to have Python 3.8 as a cached version? I could understand not having 3.7, since it's end of life, but 3.8 is maintained until October 2024, so having it in the cache as well could be nice.

**Comment by kasper93** [comment]
wine seems to crash with futex error
```
The futex facility returned an unexpected error code.
Aborted                 (core dumped) wine ./mpv.com -v --no-config
```

**Comment by ulysses4ever** [comment]
Hey @erik-bershel. I'm one of the Cabal maintainers (the Haskell build tool included in the runners). The upgrade to GHC 9.10 was premature: nothing in the ecosystem is ready for it yet. Please, roll back to GHC 9.8 or better yet 9.6, [which was declared to be a sort of an LTS by the GHC team recently](https://www.haskell.org/ghc/blog/20240521-ghc-release-priorities.html). 

Context. Currently, 9.10 is considered a bleeding edge. People can easily get it via ghcup (also included in the runners) but it's a bad default because, as @simonmichael mentions above: most software will fail to build with it. Cabal's own CI had a failure because we relied on the GHC from the runner for our "quick" jobs: they're supposed to run quickly, so we did't want to install any particular GHC and thought the the runner will provide a sensible default (https://github.com/haskell/cabal/pull/10026).

**Comment by simonmichael** [comment]
Listen to the GHC/Cabal maintainers over me, but :

- +1 for waiting on GHC 9.10
- Using ghcup's Recommended version would sound like a good policy.. except that's currently 9.4.8. Probably for some good reason, but many users will find ghcup's recommendation a bit too conservative.
- Count me as one GHC+github actions user who enjoys having 9.8 rather than 9.6.

**Comment by ulysses4ever** [comment]
Using what ghcup deems recommended would sound like the most reasonable thing to do, except ghcup has, in my opinion, a very unintuitive and too conservative strategy for updating those. For GHC, ghcup-recommended makes some sense. For cabal, ghcup-recommended version was way out of date for a long time. So, my hunch would be to use ghcup-recommended GHC and ghcup-latest cabal (note, that generally using newer cabals with older GHCs is perfectly fine, but not vice versa). I'm fairly relaxed about what particular GHC version to get (9.4//6/8 but certainly not 9.10), though, but creating a general policy is hard...

**Comment by andreasabel** [comment]
@erik-bershel wrote:
> ```shell
> available_versions=$(ghcup list -t ghc -r | grep -v "prerelease" | awk '{print $2}')
> 
> # Install 2 latest Haskell Major.Minor versions
> major_minor_versions=$(echo "$available_versions" | cut -d"." -f 1,2 | uniq | tail -n2)
> ```

I think it would make sense to preinstall `latest..recommended` (or, if they coincide, the latest 2), and then set `recommended` as the default `ghc`.  In the current situation, `ghc-9.10.1` (latest), `ghc-9.8.2`, `ghc-9.6.5` and `ghc-9.4.8` (recommended) would be available and `ghc` would point to the last of these.

**Comment by ulysses4ever** [comment]
Oh, if pre-installing several versions is an option, then what Andreas suggested looks perfect to me.

**Comment by albinahlback** [comment]
For our Ubuntu 24.04 runners, we are getting the warning `Received request to deprovision: The request was cancelled by the remote provider.` followed by a cancellation of the run when it passes more than 4 minutes. [Here](https://github.com/flintlib/flint/actions/runs/9282671570?pr=2007) is an example run.

I believe it started yesterday/today, and nothing noteworthy has really changed in our repository in the past couple of days.

**Comment by m3dwards** [comment]
I'm also getting `Received request to deprovision: The request was cancelled by the remote provider` on Ubuntu 24.04 jobs after 4:10 seconds.

[Example](https://github.com/m3dwards/bitcoin/actions/runs/9287180009/job/25555541154
)

Sometimes it's failing at 4:10 but without the deprovision message and just `The hosted runner encountered an error while running your job. (Error Type: Failure).`

[Example without deprovision message](https://github.com/m3dwards/bitcoin/actions/runs/9287459567/job/25556535514
)

I _think_ there might be something resourcey going on as I've managed to get the job to run a couple of times when giving it less to do or running with less threads.

**Comment by MichaIng** [comment]
Regarding
```
Received request to deprovision: The request was cancelled by the remote provider
```
In our case, this happens when running multiple builds for multiple architectures in parallel, where APT packages are installed and upgraded within a QEMU-emulated container. To me it looks like the above warning appears for those jobs, which are currently printing/doing the
```
Calculating upgrade...
```
of APT, while installing build dependencies and upgrading the container image within the same call (`apt-get dist-upgrade package1 package2 ...`). All other currently running jobs of the same workflow run are then cancelled as well, no matter what they are currently doing, producing the
```
The hosted runner encountered an error while running your job. (Error Type: Failure).
```
error instead.

Surely APT calculating upgrades is CPU-intense already, probably additionally with the new APT resolver, and multiplied when being QEMU-emulated. Quite an argument for the native ARM runners, which will reduce load over hosted GitHub runners massively, once users adopt. We will ASAP 🙂.

However, the same stuff worked and works well on Ubuntu 22.04 runners, and of course a proper reaction would be to limit CPU utilisation of the runner, instead of just killing it. RAM usage btw, as indicated in one of the older linked issues, is likely not the reason, since this produces different runner-side error messages, usually just crashing whichever RAM-intense step/command currently runs and/or OOM kills, but all transparent in logs.

This, unless the VM host itself runs out of RAM. Proper reaction then would be to reduce the runners assigned RAM, or the amount of available runners.

**Comment by jauderho** [comment]
I am seeing the same problem as @MichaIng but it only started showing up in the last day or so. Reverting to 22.04 for now until this is fixed.

**Comment by albinahlback** [comment]
> For our Ubuntu 24.04 runners, we are getting the warning `Received request to deprovision: The request was cancelled by the remote provider.` followed by a cancellation of the run when it passes more than 4 minutes. [Here](https://github.com/flintlib/flint/actions/runs/9282671570?pr=2007) is an example run.
> 
> I believe it started yesterday/today, and nothing noteworthy has really changed in our repository in the past couple of days.
> 
> Perhaps related to #7897.

Our runners now seems to be okay.

**Comment by Thhj9** [comment]
contributing/images/snagit-theme-github-docs.snagtheme

**Comment by FeynmanZhou** [comment]
I am a maintainer of [ORAS project](https://github.com/oras-project/oras). Will ORAS CLI still be reserved in Ubuntu 24.04 GA runner machine?

**Comment by finagolfin** [comment]
@mikhailkoliada, [Swift just added 24.04 support for the latest patch release](https://www.swift.org/download/#releases), would be good to add that to this image. Unless you meant something else by "maintenance reasons" above, other than there not being official 24.04 builds till now?

**Comment by Chalakadir** [comment]
That is best Ubuntu image

**Comment by Chalakadir** [comment]
Ubuntu image beloved

**Comment by Chalakadir** [comment]
Baay'ee bareedadha

**Comment by Chalakadir** [comment]
- ### ###

**Comment by Chalakadir** [comment]
Good friend

**Comment by Chalakadir** [comment]
### ****

**Comment by Chalakadir** [comment]
Good work Ubuntu

**Comment by Laluakku** [comment]
![Uploading fotor_1718323087600.png…]()

**Comment by Chalakadir** [comment]
***@***.***

**Comment by Chalakadir** [comment]
**My** pri

**Comment by Chalakadir** [comment]
- [ ] [](url)

**Comment by Chalakadir** [comment]
> 
Duplicate of #

**Comment by MikeMcC399** [comment]
@mikhailkoliada 

It would be great to see Firefox and Edge browsers added to the [Ubuntu 24.04 beta > Browsers and Drivers](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#browsers-and-drivers) section.

Is there any timeline available for this addition? I'm assuming that they are planned for the GA release of the runner in any case. Hopefully that is true?!

**Comment by passcod** [comment]
Will this image be available on the ARM64 runners?

**Comment by asbiin** [comment]
I've found that Ubuntu 24.04 image is not working for Azure DevOps self-hosted VMSS: https://github.com/actions/runner-images/issues/10122

**Comment by esaulpaugh** [comment]
gradle still failing with exit code 134:

https://github.com/esaulpaugh/headlong-cli/actions/runs/9684527870/job/26722612613

**Comment by Krinkle** [comment]
Tried upgrading but Firefox is missing. E.g. when using QUnit and Testem.

> ```
> Launcher headless firefox not found. Not installed?
> ```
> https://github.com/Krinkle/node-colorfactory/actions/runs/9734886433/job/26863453598

Working around by switching back to the Ubuntu 20 image for now.

**Comment by Andredreyer1** [comment]
Thanks

**Comment by w1s3one805** [comment]
com.google.android.apps.enterprise.cpanel

**Comment by MikeMcC399** [comment]
Is there any news on the availability of additional browsers, Firefox and Edge?
- See also https://github.com/actions/runner-images/issues/9848#issuecomment-2180041807, which didn't get any response so far.

**Comment by Parasos** [comment]
> ### Breaking changes
> Ubuntu 24.04 is now available for all GitHub Actions and Azure DevOps users. You can use the ubuntu 24.04 image label in your YAML to select this image.
> 
> **GitHub Actions**
> 
> ```
> jobs:
>   jobName:
>     runs-on: ubuntu-24.04
> ```
> 
> **Azure DevOps**
> 
> ```
> pool:
>    name: 'Azure Pipelines'
>    vmImage: 'ubuntu-24.04'
> ```
> 
> The image is marked as "beta" for now. It means some software can be unstable on the new platform. Also, there could be queueing issues as the capacity will be balanced only throughout the next weeks. **Please report any problems with the new image to this repository.**
> 
> ### Platforms affected
> * [x]  Azure DevOps
> * [x]  GitHub Actions
> 
> ### Runner images affected
> * [ ]  Ubuntu 20.04
> * [ ]  Ubuntu 22.04
> * [x]  Ubuntu 24.04
> * [ ]  macOS 11
> * [ ]  macOS 12
> * [ ]  macOS 13
> * [ ]  macOS 13 Arm64
> * [ ]  macOS 14
> * [ ]  macOS 14 Arm64
> * [ ]  Windows Server 2019
> * [ ]  Windows Server 2022
> 
> ### Software difference between Ubuntu 22 and Ubuntu 24
> Tool name	Ubuntu 22.04	Ubuntu 24.04	Notes
> Clang	
> * 13.*
> * 14.* (default)
> * 15.*
> 
> * 16.*
> * 17.*
> * 18.* (default)
> 
> The most recent versions are installed
> GCC / GNU C++ / GNU Fortran	
> * 9.*
> * 10.*
> * 11.*
> * 12.*
> * 13.*
> 
> * 12.*
> * 13.*
> * 14.*
> 
> The most recent versions are installed
> Kotlin	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> PHP	8.1.*	8.3.*	The most recent version are installed
> Swift	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Java	
> * 8.*
> * 11.* (default)
> * 17.*
> * 21.*
> 
> * 8.*
> * 11.*
> * 17.* (default)
> * 21.*
> 
> Default Java switched to 17.* for Ubuntu 24.04 image.
> Python	
> * 3.7.* (cached)
> * 3.8.* (cached)
> * 3.9.* (cached)
> * 3.10.* (default)
> * 3.11.* (cached)
> * 3.12.* (cached)
> 
> * 3.9.* (cached)
> * 3.10.* (default)
> * 3.11.* (cached)
> * 3.12.* (default)
> 
> Pre-cached versions currently unavailable. Default version switched to the latest one. On GitHub Actions, [actions/setup-python](https://github.com/actions/setup-python) can install any version on-flight so this change doesn't impact users
> Go	
> * 1.20.* (cached)
> * 1.21.* (default)
> * 1.22.* (cached)
> 
> * 1.20.* (cached)
> * 1.21.* (cached)
> * 1.22.* (default)
> 
> If your use-case requires using any of these versions, consider using tasks to install Go on-flight:
> * [actions/setup-go](https://github.com/actions/setup-go) (GitHub Actions)
> * [Go Tool Installer](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/go-tool?view=azure-devops) (Azure DevOps)
> 
> PyPy	
> * 3.7.* (cached)
> * 3.8.* (cached)
> * 3.9.* (cached)
> * 3.10.* (cached)
> 
> * 3.9.* (cached)
> * 3.10.* (cached)
> 
> Deprecated all the versions besides the most recent ones
> Ruby	
> * 3.0.* (default)
> * 3.1.* (cached)
> 
> * 3.2.* (default)
> 
> Pre-cached versions currently unavailable. Default version switched to the latest one.
> Node.js	
> * 16.* (cached)
> * 18.* (default)
> * 20.* (cached)
> 
> * 16.* (cached)
> * 18.* (cached)
> * 20.* (default)
> 
> If your use-case requires using any of these versions, consider using tasks to install Node.js on-flight:
> * [actions/setup-node](https://github.com/actions/setup-node) (GitHub Actions)
> * [Node.js Tool Installer](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/node-js?view=azure-devops) (Azure DevOps)
> 
> Heroku	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Leiningen	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Mono / MSBuild / NuGet	latest available	-	Software is not available for Ubuntu 24 at the moment.
> Packer / Terraform	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> R	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> SVN	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Alibaba Cloud CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Google Cloud CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Netlify CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> OpenShift CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> ORAS CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Vercel CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Bindgen / Cbindgen	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Cargo audit/clippy/outdated	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Microsoft Edge	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Microsoft Edge WebDriver	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Mozilla Firefox	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Mozilla Geckodriver	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> .NET Core SDK	
> * 6.*
> * 7.*
> * 8.*
> 
> * 8.*
> 
> Please consider using tasks to install any version on-flight:
> * [actions/setup-dotnet](https://github.com/actions/setup-dotnet) (GitHub Actions)
> * [Use .NET Core](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops) (Azure DevOps)
> 
> PostgreSQL	14.*	16.*	More recent version are installed
> MS SQL Server Client Tools	sqlcmd / SqlPackage	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> MarkdownPS Module	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Android Command Line Tools	9.0	12.0	The most recent version are installed
> Android SDK Build-tools	
> * 34.0.0
> * 33.0.0
> * 33.0.1
> * 33.0.2
> * 33.0.3
> * 32.0.0
> * 31.0.0
> 
> * 34.0.0
> 
> The most recent version are installed
> Android NDK	
> * 25.* (default)
> * 26.*
> 
> * 26.* (default)
> 
> The most recent version are installed
> Cached Docker images	
> * alpine:3.16
> * alpine:3.17
> * alpine:3.18
> * debian:10
> * debian:11
> * moby/buildkit:latest
> * node:16
> * node:16-alpine
> * node:18
> * node:18-alpine
> * node:20
> * node:20-alpine
> * ubuntu:20.04
> * ubuntu:22.04
> 
> -	Removed from the Ubuntu 24.04 image due to maintenance reasons.

**Comment by drmmr763** [comment]
Attempting to switch my pipeline to this by specifying it as the agent in my pipeline:

```
# Agent VM image name
  vmImageName: 'ubuntu-24.04'

stages:
- stage: Build
  displayName: Build stage
  variables:
    phpVersion: '8.3'
  jobs:
  - job: BuildJob
    pool:
      vmImage: $(vmImageName)
```

However when running I'm getting this in the logs:
```
Pool: [Azure Pipelines](redacted)
Image: ubuntu-24.04
Queued: Today at 12:15 PM [[manage parallel jobs](redacted)]

Acquiring an agent from the cloud:
This agent cloud does not support more specific status information./nPool provider seems slow, we notified pool provider at 7/23/2024 4:15:44 PM, still agent has not started processing it. Pool provider delay(in mins): 8.69122598666667
```

It looks like the pool is not issuing me the agent for some reason. Do I have to do something specific to activate it or there still issues with provisioning it?

**Comment by h-vetinari** [comment]
Is this available for Azure Pipeline yet? I tried this today (following the instructions in the OP), and got an immediate cancellation:
```
##[error]No config name or imagelabel provided in request
,##[error]The remote provider was unable to process the request.
Pool: Azure Pipelines
Image: ubuntu-24.04
```

**Comment by AngelicaOvesdotter** [comment]
![image](https://github.com/user-attachments/assets/94da5212-19f5-4eff-91d4-d408394cfc55)

**Comment by Damien-Chen** [comment]
Is `ubuntu-24.04` image available on Azure Pipeline ? Same question as @h-vetinari

**Comment by AngelicaOvesdotter** [comment]
![image](https://github.com/user-attachments/assets/e638db4c-cf9e-4885-85e8-5df3c639596e)

**Comment by EwoutH** [comment]
With 24.04 being out for a few months now, would it be a good time to start talking about a timeline for switching over `ubuntu-latest` to point to `ubuntu-24.04`?

**Comment by esaulpaugh** [comment]
> With 24.04 being out for a few months now, would it be a good time to start talking about a timeline for switching over `ubuntu-latest` to point to `ubuntu-24.04`?

Just as soon as gradle stops being broken https://github.com/esaulpaugh/headlong-cli/actions/runs/9993964738/job/28268178266

**Comment by erik-bershel** [comment]
Hey @esaulpaugh!
Take a look here: https://github.com/actions/runner-images/issues/9977

**Comment by erik-bershel** [comment]
Hey @Damien-Chen, @h-vetinari, and @drmmr763.

> Is ubuntu-24.04 image available on Azure Pipeline ? 

All the available agents might be found here: https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software
Unfortunately I can not provide more info regarding image availability for Azure DevOps.

**Comment by h-vetinari** [comment]
> @erik-bershel: Unfortunately I can not provide more info regarding image availability for Azure DevOps.

I appreciate the response, thanks. It's a bit confusing to me though, because the first post literally says

> **Azure DevOps**
> 
> ```
> pool:
>    name: 'Azure Pipelines'
>    vmImage: 'ubuntu-24.04'
> ```

I don't know how Azure Pipelines & GHA are separated organisationally or infrastructurally, but since their website directly links back to this repo (if one clicks on any of the "Included Software" links on the page you referenced), and it's been announced it at the top of the issue, I feel it's really up to MSFT-employees to collaborate on some form of answer/timeline for this...

**Comment by MeesterDev** [comment]
@h-vetinari is right. Indeed, this issue shows up as the second result in both Google and Bing for "azure devops ubuntu 24.04" [without quotes] and provides the yaml to use for Azure DevOps, but when actually using it, it doesn't work. However, no information can be given in the only place that heavily implies it should be working.

I understand that for a big company collaboration can get complex, but you seem to be "sort of" collaborating, which makes things quite unclear for users (or at the very least, for me).

**Comment by drmmr763** [comment]
@erik-bershel its definitely a litlte confusing as the docs here suggest that this agent should be available even though its not. Also since Azure App Services are showing PHP Version 8.0 as deprecated https://azure.microsoft.com/en-us/updates/php8support/, we are wanting to upgrade to 8.3. Upgrading to 8.3 requires the build agent to also be 8.3 for things like composer.json dependency management. 

All to say: things are not in sync here.

My temporary work around for now has been to dockerize my build steps and run PHP 8.3 and dependencies within the container by-passing the version on the `22.04` agent.

**Comment by Juan5212** [comment]
> ### Cambios importantes
> Ubuntu 24.04 ya está disponible para todos los usuarios de GitHub Actions y Azure DevOps. Puedes usar la etiqueta de imagen Ubuntu 24.04 en tu YAML para seleccionar esta imagen.
> 
> **Acciones de GitHub**
> 
> ```
> jobs:
>   jobName:
>     runs-on: ubuntu-24.04
> ```
> 
> **Azure DevOps**
> 
> ```
> pool:
>    name: 'Azure Pipelines'
>    vmImage: 'ubuntu-24.04'
> ```
> 
> La imagen está marcada como "beta" por ahora. Esto significa que algunos programas pueden ser inestables en la nueva plataforma. Además, podría haber problemas de cola, ya que la capacidad se equilibrará solo durante las próximas semanas. **Informe cualquier problema con la nueva imagen a este repositorio.**
> 
> ### Plataformas afectadas
> * [x] Azure DevOps
> * [x] Acciones de GitHub
> 
> ### Imágenes de corredores afectados
> * [ ] Ubuntu 20.04
> * [ ] Ubuntu 22.04
> * [x] Ubuntu 24.04
> * [ ] macOS 11
> * [ ] macOS 12
> * [ ] macOS 13
> * [ ] macOS 13 Arm64
> * [ ] macOS 14
> * [ ] macOS 14 Arm64
> * [ ] Servidor Windows 2019
> * [ ] Servidor Windows 2022
> 
> ### Diferencias de software entre Ubuntu 22 y Ubuntu 24
> Nombre de la herramienta	Ubuntu 22.04	Ubuntu 24.04	Notas
> Sonido metálico	
> * 13.*
> * 14.* (predeterminado)
> * 15.*
> 
> * 16.*
> * 17.*
> * 18.* (predeterminado)
> 
> Se instalan las versiones más recientes
> GCC / GNU C++ / GNU Fortran	
> * 9.*
> * 10.*
> * 11.*
> * 12.*
> * 13.*
> 
> * 12.*
> * 13.*
> * 14.*
> 
> Se instalan las versiones más recientes
> Kotlin	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> PHP	8.1.*	8.3.*	Se instala la versión más reciente
> Rápido	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Java	
> * 8.*
> * 11.* (predeterminado)
> * 17.*
> * 21.*
> 
> * 8.*
> * 11.*
> * 17.* (predeterminado)
> * 21.*
> 
> Java predeterminado cambiado a 17.* para la imagen de Ubuntu 24.04.
> Pitón	
> * 3.7.* (en caché)
> * 3.8.* (en caché)
> * 3.9.* (en caché)
> * 3.10.* (predeterminado)
> * 3.11.* (en caché)
> * 3.12.* (en caché)
> 
> * 3.9.* (en caché)
> * 3.10.* (predeterminado)
> * 3.11.* (en caché)
> * 3.12.* (predeterminado)
> 
> Las versiones almacenadas previamente en caché no están disponibles actualmente. La versión predeterminada se cambió a la más reciente. En GitHub Actions, [acciones/setup-python](https://github.com/actions/setup-python) puede instalar cualquier versión sobre la marcha, por lo que este cambio no afecta a los usuarios
> Ir	
> * 1.20.* (en caché)
> * 1.21.* (predeterminado)
> * 1.22.* (en caché)
> 
> * 1.20.* (en caché)
> * 1.21.* (en caché)
> * 1.22.* (predeterminado)
> 
> Si su caso de uso requiere el uso de alguna de estas versiones, considere usar tareas para instalar Go durante el vuelo:
> * [acciones/configuración-go](https://github.com/actions/setup-go) (Acciones de GitHub)
> * [Instalador de herramientas Go](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/go-tool?view=azure-devops) (Azure DevOps)
> 
> PyPy	
> * 3.7.* (en caché)
> * 3.8.* (en caché)
> * 3.9.* (en caché)
> * 3.10.* (en caché)
> 
> * 3.9.* (en caché)
> * 3.10.* (en caché)
> 
> Se han descontinuado todas las versiones excepto las más recientes
> Rubí	
> * 3.0.* (predeterminado)
> * 3.1.* (en caché)
> 
> * 3.2.* (predeterminado)
> 
> Las versiones almacenadas previamente en caché no están disponibles actualmente. Se cambió la versión predeterminada a la más reciente.
> Node.js	
> * 16.* (en caché)
> * 18.* (predeterminado)
> * 20.* (en caché)
> 
> * 16.* (en caché)
> * 18.* (en caché)
> * 20.* (predeterminado)
> 
> Si su caso de uso requiere el uso de alguna de estas versiones, considere usar tareas para instalar Node.js sobre la marcha:
> * [acciones/configuración-nodo](https://github.com/actions/setup-node) (Acciones de GitHub)
> * [Instalador de herramientas Node.js](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/node-js?view=azure-devops) (Azure DevOps)
> 
> Héroeku	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Leiningen	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Mono / MSBuild / NuGet	Último disponible	-	El software no está disponible para Ubuntu 24 en este momento.
> Empaquetador / Terraform	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> R	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> SVN	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Interfaz de línea de comandos de Alibaba Cloud	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Interfaz de línea de comandos de Google Cloud	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Interfaz de línea de comandos de Netlify	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Interfaz de línea de comandos de OpenShift	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> ORAS CLI	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Interfaz de línea de comandos de Vercel	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Bindgen / Cbindgen	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Auditoría de carga/recortada/desactualizada	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Microsoft Edge	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Controlador web de Microsoft Edge	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Mozilla Firefox	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Controlador Gecko de Mozilla	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Kit de desarrollo de software para .NET Core	
> * 6.*
> * 7.*
> * 8.*
> 
> * 8.*
> 
> Considere utilizar tareas para instalar cualquier versión durante el vuelo:
> * [acciones/setup-dotnet](https://github.com/actions/setup-dotnet) (Acciones de GitHub)
> * [Utilice .NET Core](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops) (Azure DevOps)
> 
> PostgreSQL	14.*	16.*	Se instalan versiones más recientes
> Herramientas de cliente de MS SQL Server	sqlcmd / Paquete SQL	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Módulo MarkdownPS	Último disponible	-	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.
> Herramientas de línea de comandos de Android	9.0	12.0	Se instala la versión más reciente
> Herramientas de compilación del SDK de Android	
> * 34.0.0
> * 33.0.0
> * 33.0.1
> * 33.0.2
> * 33.0.3
> * 32.0.0
> * 31.0.0
> 
> * 34.0.0
> 
> Se instala la versión más reciente
> Kit de desarrollo de Android	
> * 25.* (predeterminado)
> * 26.*
> 
> * 26.* (predeterminado)
> 
> Se instala la versión más reciente
> Imágenes Docker almacenadas en caché	
> * alpino:3.16
> * alpino:3.17
> * alpino:3.18
> * debian:10
> * debian:11
> * moby/buildkit:último
> * nodo:16
> * nodo:16-alpino
> * nodo:18
> * nodo:18-alpino
> * nodo:20
> * nodo:20-alpino
> * Ubuntu:20.04
> * Ubuntu:22.04
> 
> -	Eliminado de la imagen de Ubuntu 24.04 por razones de mantenimiento.

pool:
   name: 'Azure Pipelines'
   vmImage: 'ubuntu-24.04'

**Comment by EwoutH** [comment]
> With 24.04 being out for a few months now, would it be a good time to start talking about a timeline for switching over `ubuntu-latest` to point to `ubuntu-24.04`?

Are there any blocking issues on starting this process?

**Comment by MikeMcC399** [comment]
@EwoutH 

> Are there any blocking issues on starting this process?

I guess the first step would be to come out of beta status, otherwise it would be a jump directly from beta to latest.

I'm still waiting for Edge and Firefox browsers to be added to the [Ubuntu 24.04](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#browsers-and-drivers) runner.

**Comment by janbrasna** [comment]
@mikhailkoliada There are two "default" python versions in the OP overview table:

3.9.* (cached)
3.10.* **<mark>(default)</mark>**
3.11.* (cached)
3.12.* **<mark>(default)</mark>**

but it's probably just the latter that's meant to be the new default, right?

**Comment by AngelicaOvesdotter** [comment]
```AppleScript

```

**Comment by MikeMcC399** [comment]
[Ubuntu 24.04.1 LTS](https://ubuntu.com/blog/upgrade-your-desktop-ubuntu-24-04-lts) is now available.

Are there any new plans for the beta Ubuntu `24.04` that can be shared?

**Comment by MikeMcC399** [comment]
- I see that [ubuntu24/20240908.1](https://github.com/actions/runner-images/releases/tag/ubuntu24/20240908.1) updates to Ubuntu `24.04.1`

**Comment by MikeMcC399** [comment]
@erik-bershel 

> GitHub [Roadmap](https://github.com/orgs/github/projects/4247/views/1?filterQuery=is%3Aopen+-status%3A%22Q4+2022+%E2%80%93+Oct-Dec%22%2C%22Q2+2023+%E2%80%93+Apr-Jun%22%2C%22Q3+2023+%E2%80%93+Jul-Sep%22%2C%22Q4+2023+%E2%80%93+Oct-Dec%22+label%3Arunners) for `runners`. 

Thanks for the link to the roadmap which says Ubuntu 24.04 is planned to move to GA in the July - Sep 2024 Q3 quarter, of which there are now less than 3 weeks left, so I wonder if that is still feasible?

I'm still waiting for Microsoft Edge and Mozilla Firefox browsers to be added to Ubuntu 24.04 to allow cross-browser testing. If that is not at all planned, it would be good to have some advance notice.

**Comment by ijunaidm** [comment]
@MikeMcC399 - We have added Mozilla firefox browser this week, hopefully it should be available next week on ubuntu24.04 image. And regarding edge i will check and add it to the image.

**Comment by MatthewSteeples** [comment]
@MikeMcC399 See #10636 if you haven't already (for anyone else that's subscribed to this thread but not new issues)

**Comment by MikeMcC399** [comment]
@MatthewSteeples 

Thank you! I happened to click on the https://github.com/actions/runner-images/labels/Announcement label just now. It's also pinned to the top of issues.

**Comment by MasterooGrr24** [comment]
> ### Breaking changes
> Ubuntu 24.04 is now available for all GitHub Actions and Azure DevOps users. You can use the ubuntu 24.04 image label in your YAML to select this image.
> 
> **GitHub Actions**
> 
> ```
> jobs:
>   jobName:
>     runs-on: ubuntu-24.04
> ```
> 
> **Azure DevOps**
> 
> ```
> pool:
>    name: 'Azure Pipelines'
>    vmImage: 'ubuntu-24.04'
> ```
> 
> The image is marked as "beta" for now. It means some software can be unstable on the new platform. Also, there could be queueing issues as the capacity will be balanced only throughout the next weeks. **Please report any problems with the new image to this repository.**
> 
> ### Platforms affected
> * [x]  Azure DevOps[x]  GitHub Actions
> 
> ### Runner images affected
> * [ ]  Ubuntu 20.04[ ]  Ubuntu 22.04[x]  Ubuntu 24.04[ ]  macOS 11[ ]  macOS 12[ ]  macOS 13[ ]  macOS 13 Arm64[ ]  macOS 14[ ]  macOS 14 Arm64[ ]  Windows Server 2019[ ]  Windows Server 2022
> 
> ### Software difference between Ubuntu 22 and Ubuntu 24
> Tool name	Ubuntu 22.04	Ubuntu 24.04	Notes
> Clang	
> * 13.*
> * 14.* (default)
> * 15.*
> 
> * 16.*
> * 17.*
> * 18.* (default)
> 
> The most recent versions are installed
> GCC / GNU C++ / GNU Fortran	
> * 9.*
> * 10.*
> * 11.*
> * 12.*
> * 13.*
> 
> * 12.*
> * 13.*
> * 14.*
> 
> The most recent versions are installed
> Kotlin	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> PHP	8.1.*	8.3.*	The most recent version are installed
> Swift	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Java	
> * 8.*
> * 11.* (default)
> * 17.*
> * 21.*
> 
> * 8.*
> * 11.*
> * 17.* (default)
> * 21.*
> 
> Default Java switched to 17.* for Ubuntu 24.04 image.
> Python	
> * 3.7.* (cached)
> * 3.8.* (cached)
> * 3.9.* (cached)
> * 3.10.* (default)
> * 3.11.* (cached)
> * 3.12.* (cached)
> 
> * 3.9.* (cached)
> * 3.10.* (default)
> * 3.11.* (cached)
> * 3.12.* (default)
> 
> Pre-cached versions currently unavailable. Default version switched to the latest one. On GitHub Actions, [actions/setup-python](https://github.com/actions/setup-python) can install any version on-flight so this change doesn't impact users
> Go	
> * 1.20.* (cached)
> * 1.21.* (default)
> * 1.22.* (cached)
> 
> * 1.20.* (cached)
> * 1.21.* (cached)
> * 1.22.* (default)
> 
> If your use-case requires using any of these versions, consider using tasks to install Go on-flight:
> * [actions/setup-go](https://github.com/actions/setup-go) (GitHub Actions)
> * [Go Tool Installer](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/go-tool?view=azure-devops) (Azure DevOps)
> 
> PyPy	
> * 3.7.* (cached)
> * 3.8.* (cached)
> * 3.9.* (cached)
> * 3.10.* (cached)
> 
> * 3.9.* (cached)
> * 3.10.* (cached)
> 
> Deprecated all the versions besides the most recent ones
> Ruby	
> * 3.0.* (default)
> * 3.1.* (cached)
> 
> * 3.2.* (default)
> 
> Pre-cached versions currently unavailable. Default version switched to the latest one.
> Node.js	
> * 16.* (cached)
> * 18.* (default)
> * 20.* (cached)
> 
> * 16.* (cached)
> * 18.* (cached)
> * 20.* (default)
> 
> If your use-case requires using any of these versions, consider using tasks to install Node.js on-flight:
> * [actions/setup-node](https://github.com/actions/setup-node) (GitHub Actions)
> * [Node.js Tool Installer](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/node-js?view=azure-devops) (Azure DevOps)
> 
> Heroku	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Leiningen	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Mono / MSBuild / NuGet	latest available	-	Software is not available for Ubuntu 24 at the moment.
> Packer / Terraform	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> R	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> SVN	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Alibaba Cloud CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Google Cloud CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Netlify CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> OpenShift CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> ORAS CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Vercel CLI	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Bindgen / Cbindgen	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Cargo audit/clippy/outdated	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Microsoft Edge	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Microsoft Edge WebDriver	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Mozilla Firefox	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Mozilla Geckodriver	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> .NET Core SDK	
> * 6.*
> * 7.*
> * 8.*
> 
> * 8.*
> 
> Please consider using tasks to install any version on-flight:
> * [actions/setup-dotnet](https://github.com/actions/setup-dotnet) (GitHub Actions)
> * [Use .NET Core](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops) (Azure DevOps)
> 
> PostgreSQL	14.*	16.*	More recent version are installed
> MS SQL Server Client Tools	sqlcmd / SqlPackage	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> MarkdownPS Module	latest available	-	Removed from the Ubuntu 24.04 image due to maintenance reasons.
> Android Command Line Tools	9.0	12.0	The most recent version are installed
> Android SDK Build-tools	
> * 34.0.0
> * 33.0.0
> * 33.0.1
> * 33.0.2
> * 33.0.3
> * 32.0.0
> * 31.0.0
> 
> * 34.0.0
> 
> The most recent version are installed
> Android NDK	
> * 25.* (default)
> * 26.*
> 
> * 26.* (default)
> 
> The most recent version are installed
> Cached Docker images	
> * alpine:3.16
> * alpine:3.17
> * alpine:3.18
> * debian:10
> * debian:11
> * moby/buildkit:latest
> * node:16
> * node:16-alpine
> * node:18
> * node:18-alpine
> * node:20
> * node:20-alpine
> * ubuntu:20.04
> * ubuntu:22.04
> 
> -	Removed from the Ubuntu 24.04 image due to maintenance reasons.

**Comment by sham25678** [comment]
jobs:
  jobName:
    runs-on: ubuntu-24.04
